jupyter_server/extension/application.py

```diff
@@ -214,47 +214,75 @@ def _prepare_templates(self):
         self.initialize_templates()
 
     @staticmethod
-    def initialize_server():
+    def initialize_server(argv=[], **kwargs):
         """Get an instance of the Jupyter Server."""
         # Get a jupyter server instance
-        serverapp = ServerApp()
+        serverapp = ServerApp(**kwargs)
         # Initialize ServerApp config.
         # Parses the command line looking for 
         # ServerApp configuration.
-        serverapp.initialize()
+        serverapp.initialize(argv=argv)
         return serverapp
 
-    def initialize(self, serverapp, argv=None):
-        """Initialize the extension app."""
+    def initialize(self, serverapp, argv=[]):
```

Comment:
Lint: Using a mutable arg as default value. Several other such cases in the PR as well.

Reply:
Good point. Every call will get the same instance of the empty list, so if somebody modifies the list to be non-empty...

Reply:
Hm, right. That could cause some inheritance problems. I'll submit a fix in a later PR.